### PR TITLE
Menu fixes.

### PR DESCRIPTION
**Pull request by tclaar**
Fixed: 
- buttons still listening for a mouse click even after being removed
- button text/respawn menu items not appearing

### CODE REVIEW

File: js/classes/entities/foreground/ui/button.js [hunk 1]
```diff
@@ -32,7 +31,8 @@ class Button {
                     this.removeFromWorld = true;
                 }
             }
-        });
+        }
+        document.body.addEventListener("click", this.listener);
```
Comment (calebkrauter) [inline js/classes/entities/foreground/ui/button.js:35]: I am curious to know why this change makes a difference. Can you explain?

File: js/classes/entities/foreground/ui/button.js [hunk 2]
```diff
@@ -61,8 +61,8 @@ class Button {
         const buttonTextSize = CTX.measureText(this.text);
         CTX.fillText(
             this.text, this.pos.x + (this.size.x - buttonTextSize.width) / 2,
-            this.pos.y + this.size.y - (this.size.y - buttonTextSize.emHeightAscent
-                + buttonTextSize.emHeightDescent) / 2
+            this.pos.y + this.size.y - (this.size.y - buttonTextSize.actualBoundingBoxAscent
```
Comment (calebkrauter) [inline js/classes/entities/foreground/ui/button.js:64]: I appreciate this refactor. Is Ascent and Descent used to imply that the bounding box has some y value in the negative or positive direction that we are using? Also why is it "actual bounding box" and not just "bounding box"? What would be the difference? Wondering if further refactoring would make sense or not. I'm a little confused and flexible on this, what are your thoughts?

**Review by calebkrauter — APPROVED** [review]
Thanks for fixing this!

File: js/classes/entities/foreground/ui/button.js (hunk 1 above)
Comment (tclaar) [reply js/classes/entities/foreground/ui/button.js:35]: Before, I was assigning `this.listener` the return value of `addEventListener`, which makes no sense because `addEventListener` returns `undefined`. This was a problem because `this.listener` was being passed to `removeEventListener`, which didn't do anything because `removeEventListener` expects the listener function as its second argument. That's why `this.listener` is assigned the listener function itself now.

File: js/classes/entities/foreground/ui/button.js (hunk 2 above)
Comment (tclaar) [reply js/classes/entities/foreground/ui/button.js:64]: `buttonTextSize` is a `TextMetrics` object returned by `CTX.measureText(this.text)`, and `actualBoundingBoxAscent`, etc. are properties of this object. `boundingBox` is not a member of `TextMetrics`. As far as why they're named that way, you'd have to ask whoever wrote the API. You can read here for some more info: https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics.
